x-pack/platform/plugins/shared/streams/server/routes/streams/significant_events/read_significant_events.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AggregationsDateHistogramAggregate } from '@elastic/elasticsearch/lib/api/types';
+import { IScopedClusterClient } from '@kbn/core/server';
+import { buildEsQuery } from '@kbn/es-query';
+import { ChangePointType } from '@kbn/es-types/src';
+import { SignificantEventsGetResponse } from '@kbn/streams-schema';
+import { get, isArray, isEmpty } from 'lodash';
+import { AssetClient } from '../../../lib/streams/assets/asset_client';
+
+export async function readSignificantEvents(
+  params: { name: string; from: Date; to: Date; bucketSize: string },
+  dependencies: {
+    assetClient: AssetClient;
+    scopedClusterClient: IScopedClusterClient;
+  }
+): Promise<SignificantEventsGetResponse> {
+  const { assetClient, scopedClusterClient } = dependencies;
+  const { name, from, to, bucketSize } = params;
+
+  const assetQueries = await assetClient.getAssetLinks(name, ['query']);
+  if (isEmpty(assetQueries)) {
+    return [];
+  }
+
+  const searchRequests = assetQueries.flatMap((asset) => {
```

Comment:
should we do this in ESQL instead? Since very recently, `CHANGE_POINT` is also an ESQL command

Reply:
Maybe a little too early? I'm not sure if it's available outside of snapshot builds, it has a limit of 1000 values and not sure if we can already group by values (which might not matter for now, but will later). There's also no _msearch equivalent for ES|QL queries. We don't need it, but just a couple of things that make me feel like it's too early.

Reply:
No strong opinion, but based on the early nature of this app and the way we want to go with ESQL, it might be good to put pressure on it where possible, to highlight problems.

Reply:
@kdelemme do you mind having a look if ES|QL is available in non-snapshot builds? if so, let's see if we can use CHANGE_POINT. If we go down that route, let's also built in a limiter (pLimit is a common way to do this) to make sure we only do max 20 ES|QL requests in parallel.